### PR TITLE
HYPERFLEET-700 - refactor: extract WithMetrics and AlwaysAck wrappers

### PR DESCRIPTION
**Pull request by Mischulee**
## Summary
Replaces the `CreateHandler` callback pattern with composable decorator functions `WithMetrics` and `AlwaysAck`, extracting metrics recording and ACK logic out of the executor into wrappers.

- Ticket: [HYPERFLEET-700](https://redhat.atlassian.net/browse/HYPERFLEET-700)

## Test Plan

- [x] Unit tests added/updated
- [x] `make test-all` passes
- [x] `make lint` passes
- [ ] Helm chart changes validated with `make test-helm` (if applicable)
- [ ] Deployed to a development cluster and verified (if Helm/config changes)
- [ ] E2E tests passed (if cross-component or major changes)

## Summary by CodeRabbit

* **Refactor**
  * Metrics collection and event acknowledgement are now separate, composable handler wrappers; core execution no longer directly performs instrumentation or forces ACK behavior.
  * Handler abstraction was reworked so metrics and ACKs are applied externally for clearer separation and safer instrumentation.

* **Tests**
  * Added tests for metrics across outcomes, always-ACK behavior, and panic/edge-case handling in the wrapper utilities.

### CODE REVIEW

File: internal/executor/handler.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		result, err := h(ctx, evt)
+		duration := time.Since(start)
+
+		resultForMetrics := result
+		if err != nil && (result == nil || result.Status == StatusSuccess) {
+			resultForMetrics = &ExecutionResult{Status: StatusFailed}
+		}
+
+		func() {
+			defer func() { _ = recover() }() //nolint:errcheck
```
Comment (rafabene) [inline internal/executor/handler.go]: **Category:** Bug

The recovered panic on line 34 is silently discarded. If `recordMetrics` ever panics (e.g., nil pointer from a future metric change), there will be zero evidence in logs — making it very hard to diagnose.

Consider logging the recovered value:

```go
defer func() {
    if r := recover(); r != nil {
        // Use a lightweight log or fmt.Fprintf(os.Stderr, ...) to avoid
        // depending on the executor's logger in this utility function
        fmt.Fprintf(os.Stderr, "panic in metrics recording (recovered): %v\n", r)
    }
}()
```

Or, since the executor's logger isn't available here, you could accept a `logger.Logger` parameter in `WithMetrics`, or use `slog.Error()` directly — whichever fits the team's preference. The key point is: don't swallow the panic silently.

Comment (rafabene) [inline internal/executor/handler.go]: **Category:** Standards

The `//nolint:errcheck` directive here is unnecessary — `recover()` returns `interface{}`, not `error`, so `errcheck` wouldn't flag it. Same on line 47: the blank identifiers `_, _` already satisfy `errcheck`.

If you keep the directives, they should include a justification comment per the linting standard. But in both cases, the cleaner fix is to simply remove them:

```go
// line 34: recover returns interface{}, not error — no nolint needed
defer func() { _ = recover() }()

// line 47: blank identifiers already suppress errcheck
_, _ = h(ctx, evt)
```

File: internal/executor/executor_test.go [hunk 2]
```diff
@@ -957,6 +955,141 @@ func TestCreateHandler_NilMetricsRecorder(t *testing.T) {
 	}, "handler with nil MetricsRecorder should not panic")
 }
 
+// TestWithMetrics_RecordsMetrics verifies WithMetrics records the correct metric status
+// and passes the result through
+func TestWithMetrics_RecordsMetrics(t *testing.T) {
+	tests := []struct {
```
Comment (pnguyen44) [inline internal/executor/executor_test.go:961]: Add a test that verifies metrics recording panics are recovered.

File: internal/executor/handler.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		return result, err
+	}
+}
+
+// AlwaysAck wraps a HandlerFunc into a broker compatible handler that always returns nil,
+// preventing infinite retry loops for non-recoverable errors.
+// Execution errors are recorded by WithMetrics and logged inside the handler.
+func AlwaysAck(h HandlerFunc) func(ctx context.Context, evt *event.Event) error {
+	return func(ctx context.Context, evt *event.Event) error {
+		_, _ = h(ctx, evt) //nolint:errcheck
```
Comment (ciaranRoche) [inline internal/executor/handler.go]: This is probably fine in practice, but we just need to be aware that if any errors are not logged before we hit this point, they will vanish. 
Worth adding a doc comment here making this contract explicit to ensure callers log any errors before `AlwaysAck`

**Review by ciaranRoche — COMMENTED** [review]
This refactor is solid, good to go when the small feedback is addressed

File: internal/executor/handler.go (hunk 1 above)
Comment (Mischulee) [reply internal/executor/handler.go]: This repo has `check-blank: true` set in `golangci.yml`, which causes errcheck to treat explicit `_ ` assignments as lint violations. Removing this breaks lint. 
I added a justification comment per the linting standard as you suggested.

Comment (Mischulee) [reply internal/executor/handler.go]: Makes sense. I have added a `logger.Logger` param to `WithMetrics` and log the recovered panic with `log.Errorf` to make it consistent with codebase.

File: internal/executor/executor_test.go (hunk 2 above)
Comment (Mischulee) [reply internal/executor/executor_test.go:961]: Test added.

File: internal/executor/handler.go (hunk 3 above)
Comment (Mischulee) [reply internal/executor/handler.go]: Comment added.

File: internal/executor/handler.go (hunk 1 above)
Comment (rafabene) [reply internal/executor/handler.go]: Good catch — I missed that `check-blank: true` is enabled in this repo. The nolint directives are needed then. Adding the justification comments is the right approach, thanks.
